vmdb/spec/controllers/catalog_controller_spec.rb

```diff
@@ -77,53 +77,48 @@
     end
   end
 
-  context "#ot_name_and_description_edit" do
-    it "Orchestration Template name and description are edited" do
+  context "#ot_edit" do
+    before(:each) do
       controller.instance_variable_set(:@sb, {})
       controller.instance_variable_set(:@_params, :button => "save")
-      controller.instance_variable_set(:@_response, ActionController::TestResponse.new)
-      ot = FactoryGirl.create(:orchestration_template)
-      controller.instance_variable_set(:@record, ot)
-      new_name = "New Name"
-      new_description = "New Description"
-      new_content = "New Content"
-      edit = {
+      controller.should_receive(:render)
+      @new_name = "New Name"
+      @new_description = "New Description"
+      @new_content = "New Content"
+      session[:edit] = {
         :new    => {
-          :name        => new_name,
-          :description => new_description},
-        :key    => "ot_edit__#{ot.id}",
-        :rec_id => ot.id,
+          :name        => @new_name,
+          :description => @new_description},
       }
-      controller.params.merge!(:id => ot.id, :template_content => new_content)
-      controller.instance_variable_set(:@edit, edit)
-      session[:edit] = edit
+    end
+
+    after(:each) do
+      expect(response.status).to eq(200)
+      assigns(:edit).should be_nil
+    end
+
+    it "Orchestration Template name and description are edited" do
+      ot = FactoryGirl.create(:orchestration_template)
+      controller.instance_variable_set(:@record, ot)
+      controller.params.merge!(:id => ot.id, :template_content => @new_content)
+      session[:edit][:key] = "ot_edit__#{ot.id}"
```

Comment:
I need the ot.id here and ot is created in the "it ..."  blocks (each time a different type of ot).